test/test_list.rb

```diff
@@ -931,3 +931,59 @@ def test_exception_on_wrong_position
     end
   end
 end
+
+class ListConsistencyChecksAndRepairTest < ActsAsListTestCase
+  def setup
+    setup_db
+    (1..10).each { |counter| DefaultScopedMixin.create!({pos: counter}) }
+  end
+
+  def test_does_list_have_duplicates
+    assert_equal false, DefaultScopedMixin.first.does_list_have_duplicates?
+
+    # Create a new item and place it at an existing position
+    # We need to disable callbacks in order to accomplish this
+    DefaultScopedMixin.acts_as_list_no_update {DefaultScopedMixin.create(pos: 2)}
+    
+    assert_equal true, DefaultScopedMixin.first.does_list_have_duplicates?
+  end
+
+  def test_repair_duplicate_positions_in_list
+    # Create a few duplicate positions
+    # We need to disable callbacks in order to accomplish this
+    DefaultScopedMixin.acts_as_list_no_update {
+      [3, 4, 5, 3, 8].each { |counter| DefaultScopedMixin.create!({pos: counter}) }
+    }
+
+    # Make sure that the list has duplicates
+    assert_equal true, DefaultScopedMixin.first.does_list_have_duplicates?
+
+    # Fix the list
+    DefaultScopedMixin.first.repair_duplicate_positions_in_list
```

Comment:
I think the repair code would be better attached to the class rather than the instance. We can still call it from the instance.

Reply:
It would be `DefaultScopedMixin.repair_duplicate_positions_in_list(DefaultScopedMixin.first)` instead?